src/Versions.cs

```diff
@@ -64,11 +64,17 @@ internal IEnumerable<ApiVersion> GetDeprecatedVersions(ApiVersion? from, ApiVers
     /// Currently just returns the latest version
     /// </remarks>
     /// <returns>array with a single element, the currentVersion</returns>
-    internal IEnumerable<ApiVersion> GetSupportedVersions(ApiVersion? from, ApiVersion? until)
+    private IEnumerable<ApiVersion> GetSupportedVersions(ApiVersion? from, ApiVersion? until)
     {
-        if (until is not null && (from >= until))
+        if (until is not null)
         {
-            throw new InvalidOperationException($"The from value ({from}) has to be bigger than the until version ({until})");
+            switch (options.UntilInclusive)
```

Comment:
I think we can simplify, because if from is > until it is always wrong.

```csharp

if(from > until) {
  // some error saying that from can never be bigger than until for the same action
}
if(!options.UntilInclusive && from == until){
   // error, from can't be the same as until unless you enable "UntilInclusive"
}

```

With better error messages, obviously.

Reply:
I'll change it to this next Monday, just put away my stuff.

I'll also take a very brief look about testing using the `BetterVersioningConventionBuilder`, but those tests might just become too complex to write and maintain unfortunately.